### ISSUE TITLE
Error fetching POST  http://127.0.0.1:3210/api/push_config 500 Internal Server Error

### ISSUE THREAD

**Issue by xjq284** [a16z-infra/ai-town#227]
Diffing local code and deployment state
Analyzing and deploying source code...
✖ Error: Unable to push deployment config to http://127.0.0.1:3210
Error fetching POST  http://127.0.0.1:3210/api/push_config 500 Internal Server Error: InternalServerError: Your request couldn't be completed. Try again later.

node v21.1.0

**Comment by 61cygni** [comment]
This can happen when you run local-convex-server in a different directory
than npm-run-dev and there is a version mismatch.

So either run both in the same directory (the git root dir), or use 'nvm
use 20' in both directories.
.m

**Comment by abhavk** [comment]
Hi! I have a similar error despite matching node versions on both the server and npm run dev. 

output from running `npm run dev`: 

<img width="565" alt="image" src="https://github.com/a16z-infra/ai-town/assets/12238572/14283fda-29db-41df-939b-235762103906">

----- 

output from `./convex-local-server`:

<img width="560" alt="image" src="https://github.com/a16z-infra/ai-town/assets/12238572/729503f6-1da6-4df2-b97c-7b19d0830a46">

-----

`node` version is `v20.10.0` on both consoles and both are in the same directory (`ai-town`). Saw no failures on installation.

What am I doing wrong?

**Comment by 61cygni** [comment]
This is the same problem. You can also try running "nvm use 20" in both
directories to ensure they're using the same version.

**Comment by abhavk** [comment]
Nope, tried after running `nvm use 20` on both terminals (they were already running in the same directory, just two terminals). 

Have you seen this error for any other reason?

**Comment by 61cygni** [comment]
that's really odd. The error is very straightforward.

Another option is just to install via Pinokio which should "just work".

https://pinokio.computer/

.m

**Comment by abhavk** [comment]
Wow, got it to work on `18`. It seems to fail with node v20 and 21 (same error as above). Some notes:

1. On `node 18` it seems to work fine, although agents take some time to get moving. 
2. On `node 19` it starts up fine but I see errors repeatedly on `agent:agentDoSomething`. I can post more if that is helpful. 

----- 

P.S. Laptop going brrr and so cool to see the agents chatting, thanks for making this available! 🙂

First conversation b/w Stella and Pete is amazing: "Stella, you are lost and blinded by the devil's deceit. Turn away from your wicked ways and seek the true path of righteousness. The love of God is the only way to find true happiness and fulfillment in life."

**Comment by 61cygni** [comment]
Oh thanks for the update on this. Any chance you can submit a PR on the
readme with this? The situation seems to come up a lot.

**Comment by abhavk** [comment]
cool ill do that tomorrow, want me to include the comment about v19?

**Comment by abhavk** [comment]
#228 submitted
